analyses/mwi/dryspells_trigger/exploration/ecmwf_highres_compare_aggr.md

```diff
@@ -0,0 +1,365 @@
+        ax.xaxis.label.set_size(16)
+
+        plt.title(f'{val_col.replace("-"," minus ")}', fontdict={
+                  'size': 20})
+        plt.grid(linestyle='--', alpha=0.5)
+        ds_patch = mpatches.Patch(color=ds_color, label="dry spell")
+        nods_patch = mpatches.Patch(color=no_ds_color, label="no dry spell")
+
+        ax.set_xlim(-20,55)
+#         ax.set_xlim(-40,35)
+
+        plt.legend(handles=[ds_patch, nods_patch])
+    fig.suptitle("Difference in forecasted precipitation at leadtime=3 for different aggregation methods",fontsize=30)
+    fig.tight_layout(rect=[0, 0.03, 1, 0.98])
+```
+
+```python
+meth_comb=[[m,"cds_unrounded_center"] for m in ["api_center","api_weightavg","api_alltouched"]]
+plt_div_bar(meth_comb,colp_num=3)
+```
+
+#### Boxplot
+Only 20 datapoints per boxplot, so boxplot might not be the best method
+
+```python
+plot=alt.Chart().mark_boxplot(size=50).encode(
+    x=alt.X('method:N',sort=df_comb.method.unique()),
+    y=alt.Y(stat_col_for,scale=alt.Scale(zero=False),title="Forecasted precipitation"),
+#     facet=alt.Facet(column="month_name:N",sort=df_comb.month_name.unique())
+).properties(width=400,height=400)
+plot.facet(column=alt.Column("month_name:N",sort=df_comb.month_name.unique(),title="Month"),row=alt.Row("leadtime:N",sort=[2,3,4]),data=df_comb[df_comb.leadtime.isin([2,3,4])],
+           title=["Distribution of forecasted precipitation for different methods, facetted by leadtime and month", "Each boxplot only contains 20 datapoints"])
+```
+
+#### Histogram
+Stacked based on occurrence of dry spell    
+Due to so little observations it is not really a distrubtion which makes it hard to compare
+
+
+For some unexplainable reason the text is not readable at all. Almost seems several layers are printed on top of each other
+
+```python
+#is stacked!
+df_comb_med["ds_str"]=np.where(df_comb_med.dry_spell==1,"yes","no")
+histo = alt.Chart().mark_bar().encode(
+    x=alt.X("mean_ADM1_PCODE",bin=alt.Bin(step=10),title="Forecasted precipitation"),
+    y="count()",
+    color=alt.Color('ds_str:N',scale=alt.Scale(range=[no_ds_color,ds_color]),legend=alt.Legend(title="Dry spell")),
+)
+med_line = alt.Chart().mark_rule(color="purple").encode(
+    x='median(mean_ADM1_PCODE)')
+#TODO: for some reason text not readable, no idea why..
+med_text = alt.Chart().mark_text(align='left', dx=90, dy=-130,fontSize=15,color="purple").encode(
+    text="label:N",
+).transform_calculate(label="Median: " + alt.datum.median_for)
+
+(histo+med_line+med_text).facet(column=alt.Column("month_name:N",sort=df_comb_med.month_name.unique(),title="Month"), row=alt.Row('method:N',sort=df_comb_med.method.unique()),
+                                       data=df_comb_med[df_comb_med.leadtime==3],title="Stacked histogram of forecasted precipitation for leadtime=3, facetted by month and aggregation method")
+```
+
+```python
+# #select methods for plotting of slides
```

Comment:
Can these be removed?

Reply:
I would keep them since they are used in slide, so to make sure everything is reproducible.

Reply:
Then I would leave then uncommented maybe